common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -347,6 +379,12 @@ public Timeout newTimeout(TimerTask task, long delay, TimeUnit unit) {
         if (unit == null) {
             throw new NullPointerException("unit");
         }
+        long pendingTimeoutsCount = pendingTimeouts.incrementAndGet();
+        if (maxPendingTimeouts > 0 && pendingTimeoutsCount >= maxPendingTimeouts) {
+            throw new TooManyPendingTimeoutsException("Number of pending timeouts ("
```

Comment:
It now throws RejectedExecutionException and I have fixed the logic to only increment if we have not reached limit.